core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/internal/CarbonCoreActivator.java

```diff
@@ -69,13 +70,43 @@ public void start(BundleContext context) throws Exception {
                  System.getProperty("user.language") + "-" + System.getProperty("user.country") +
                  ", " + System.getProperty("user.timezone"));
 
-        Security.addProvider(new BouncyCastleProvider());
+        String cryptoProviderIdentifier = getPreferredJceProviderIdentifier();
+        String cryptoProviderClass = getPreferredJceProviderClass(cryptoProviderIdentifier);
+        Security.addProvider((Provider) Class.forName(cryptoProviderClass).getDeclaredConstructor().newInstance());
         if(log.isDebugEnabled()){
-            log.debug("BouncyCastle security provider is successfully registered in JVM.");
+            log.debug(cryptoProviderClass + " security provider is successfully registered in JVM.");
         }
     }
 
     public void stop(BundleContext context) throws Exception {
         dataHolder.setBundleContext(null);
     }
+
+    /**
+     * This method returns the preferred JCE provider identifier to be used.
+     *
+     * @return jce provider identifier name
+     */
+    private static String getPreferredJceProviderIdentifier() {
+        String provider = System.getProperty("security.jce.provider");
```

Comment:
Shall we move the property name to a constant as this is getting reused?

Reply:
Done with https://github.com/wso2/carbon-kernel/pull/3550/commits/baa7d216f60d63e60ef5519cb74b0b58da8d4498